arangod/Transaction/Methods.cpp

```diff
@@ -565,8 +565,30 @@ bool transaction::Methods::sortOrs(
     }
 
     auto conditionData = static_cast<ConditionData*>(parts[i].data);
-    root->addMember(conditionData->first);
-    usedIndexes.emplace_back(conditionData->second);
+    bool haveSeen = false;
+
+    if (!usedIndexes.empty()) {
+      // try to find duplicate condition parts, and only return each
+      // unique condition part once
+      try {
+        std::string const conditionString = conditionData->first->toString();
+        for (size_t j = 0; j < usedIndexes.size(); ++j) {
```

Comment:
This will be quadratic in the number of condition parts. I doubt it will be common for this to be a bottleneck, but we may want to make a note somewhere in case we run into slowdown in the future with big customer filters. Can always go back and write some kind of special handling for big conditions in later.

Reply:
we could fill an unordered_set with a pair of IndexHandle, conditionString and check for membership.
This will make it O(1). WDYT?

Reply:
Yes, that would fix the complexity. I just wonder if it's overkill for small conditions.

Reply:
Oh, sorry, I mismatched my brackets. Comment withdrawn.